dev-tools/xtask/src/virtual_hardware.rs

```diff
@@ -0,0 +1,834 @@
+    Destroy,
+}
+
+#[derive(clap::Args)]
+#[group(multiple = true)]
+pub struct Pxa {
+    /// The first IP address your Oxide cluster can use.
+    ///
+    /// Requires `pxa-end`.
+    #[clap(long = "pxa-start", requires = "end")]
+    start: Option<String>,
+
+    /// The last IP address your Oxide cluster can use
+    ///
+    /// Requires `pxa-start`.
+    #[clap(long = "pxa-end", requires = "start")]
+    end: Option<String>,
+}
+
+#[derive(Parser)]
+pub struct Args {
+    /// The directory in which virtual devices are stored
+    #[clap(long, default_value = "/var/tmp")]
+    vdev_dir: Utf8PathBuf,
+
+    #[command(subcommand)]
+    command: Commands,
+}
+
+static NO_INSTALL_MARKER: &'static str = "/etc/opt/oxide/NO_INSTALL";
+const GB: u64 = 1 << 30;
+const VDEV_SIZE: u64 = 20 * GB;
+
+const ARP: &'static str = "/usr/sbin/arp";
+const DLADM: &'static str = "/usr/sbin/dladm";
+const IPADM: &'static str = "/usr/sbin/ipadm";
+const MODINFO: &'static str = "/usr/sbin/modinfo";
+const MODUNLOAD: &'static str = "/usr/sbin/modunload";
+const NETSTAT: &'static str = "/usr/bin/netstat";
+const PFEXEC: &'static str = "/usr/bin/pfexec";
+const PING: &'static str = "/usr/sbin/ping";
+const SWAP: &'static str = "/usr/sbin/swap";
+const ZFS: &'static str = "/usr/sbin/zfs";
+const ZLOGIN: &'static str = "/usr/sbin/zlogin";
+const ZPOOL: &'static str = "/usr/sbin/zpool";
+const ZONEADM: &'static str = "/usr/sbin/zoneadm";
+
+const SIDECAR_LITE_COMMIT: &'static str =
+    "e3ea4b495ba0a71801ded0776ae4bbd31df57e26";
+const SOFTNPU_COMMIT: &'static str = "dbab082dfa89da5db5ca2325c257089d2f130092";
+const PXA_MAC_DEFAULT: &'static str = "a8:e1:de:01:70:1d";
+
+const PXA_WARNING: &'static str = r#"  You have not set up the proxy-ARP environment variables
+  PXA_START and PXA_END. These variables are necessary to allow
+  SoftNPU to respond to ARP requests for the portion of the
+  network you've dedicated to Omicron.
+  You must either destroy / recreate the Omicron environment
+  with these variables or run `scadm standalon add-proxy-arp`
+  in the SoftNPU zone later"#;
+
+pub fn run_cmd(args: Args) -> Result<()> {
+    if Utf8Path::new(NO_INSTALL_MARKER).exists() {
+        bail!("This system has the marker file {NO_INSTALL_MARKER}, aborting");
+    }
+
+    let metadata = crate::load_workspace()?;
+    let workspace_root = metadata.workspace_root;
+    let sled_agent_config =
+        workspace_root.join("smf/sled-agent/non-gimlet/config.toml");
+    let npu_zone = workspace_root.join("out/npuzone/npuzone");
+
+    match args.command {
+        Commands::Create {
+            physical_link,
+            promiscuous_filter_off,
+            gateway_ip,
+            gateway_mac,
+            pxa,
+            pxa_mac,
+        } => {
+            let physical_link = if let Some(l) = physical_link {
+                l
+            } else {
+                default_physical_link()?
+            };
+
+            println!("creating virtual hardware");
```

Comment:
Compare with: 
https://github.com/oxidecomputer/omicron/blob/d7d70d8366ee106d49cc0c7bfae2c6aa499ecf3b/tools/create_virtual_hardware.sh#L86-L93

(I did not include a `SOFTNPU_MODE == "zone"` check here, because it doesn't seem like anyone is using that? Happy to add it back if there are legitimate users)

Reply:
I see that a4x2 is using this in a few spots. I've replaced it with a `Scope` argument in:

https://github.com/oxidecomputer/omicron/pull/5423/commits/f67846b47e45584cde80925cae654f9ec4c147ee